react/components/form/ArrayFieldTemplate.js

```diff
@@ -27,38 +60,59 @@ class ArrayFieldTemplate extends Component {
     })
   }
 
+  handleSortStart = ({ node }) => {
+    console.log(node)
+    this.setState({
+      openedItem: -1,
+      sorting: true,
+    })
+  }
+
+  handleSortEnd = ({ oldIndex, newIndex }) => {
+    const { items } = this.props
+    const [{ onReorderClick }] = items
+
+    onReorderClick(oldIndex, newIndex)
+
+    this.setState({
+      sorting: false,
+    })
+  }
+
   render() {
     const { items, onAddClick, canAdd, schema } = this.props
-    const { openedItem } = this.state
+    const { openedItem, sorting } = this.state
 
     return (
-      <div>
-        {items.map(element => (
-          <Fragment
-            key={element.index}
-          >
-            <ArrayFieldTemplateItem
-              schema={schema}
-              isOpen={openedItem === element.index}
-              onOpen={this.handleOpen(element.index)}
-              onClose={this.handleClose}
-              {...element}
-            />
-            <hr style={{ height: 1, backgroundColor: '#D8D8D8', border: 'none' }} />
-          </Fragment>
-        ))}
-        <div className="pt4 tc">
+      <Fragment>
+        <ArrayList
+          items={items}
+          sorting={sorting}
+          schema={schema}
+          openedItem={openedItem}
+          onOpen={this.handleOpen}
+          onClose={this.handleClose}
+          onSortStart={this.handleSortStart}
+          onSortEnd={this.handleSortEnd}
+          helperClass="accordion-item--dragged"
+          distance={5}
+          lockAxis="y"
+          lockToContainerEdges
+          getHelperDimensions={getHelperDimensions}
+          useDragHandle
+        />
+        <div className="pt4">
           {canAdd && (
-            <Button
-              variation="secondary"
-              size="small"
+            <button
```

Comment:
Usar o botão secondary do styleguide?

Reply:
eu tirei o do styleguide porque eu to fazendo um `truncate` em parte do texto do botão, ai quando o nome é muito grande não da pra ver. ai o que eu fiz foi botar um `title` pra no hover o usuário ver o nome completo, e no botão do styleguide ele não passa a prop de title pra o `button`

Reply:
@lucasecdb vai fazer uma PR no styleguide pra resolver isso

Reply:
mandei aqui vtex/styleguide#222